statistics/handle/handle.go

```diff
@@ -831,7 +836,7 @@ func (h *Handle) columnStatsFromStorage(reader *statsReader, row chunk.Row, tabl
 		// 4. loadAll is false.
 		notNeedLoad := h.Lease() > 0 &&
 			!isHandle &&
-			(col == nil || !col.IsLoaded() && col.LastUpdateVersion < histVer) &&
+			(col == nil || !col.IsHistogramLoaded() && col.LastUpdateVersion < histVer) &&
```

Comment:
If all the samples are in TopN and histogram is empty, `col.Histogram.notNullCount()` is zero, i.e., `col.IsHistogramLoaded()` is false. Even if all the column stats have been loaded, we will enter `notNeedLoad` branch in that case. Maybe `IsLoaded` or `IsNecessaryLoaded` is more suitable here.

Reply:
updated.